test/shared/configUtils.test.ts

```diff
@@ -5,74 +5,89 @@
  * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
  */
 
+/* eslint-disable @typescript-eslint/no-unsafe-assignment */
+/* eslint-disable @typescript-eslint/no-unsafe-call */
+
 import { expect } from 'chai';
 import { Workspace } from '@lwc/lwc-dev-server';
-import { Config, ConfigAggregator } from '@salesforce/core';
+import { Config, ConfigAggregator, Connection } from '@salesforce/core';
 import { TestContext } from '@salesforce/core/testSetup';
 import { CryptoUtils } from '@salesforce/lwc-dev-mobile-core';
 import {
   ConfigUtils,
   LOCAL_DEV_SERVER_DEFAULT_PORT,
   LOCAL_DEV_SERVER_DEFAULT_WORKSPACE,
+  LocalWebServerIdentityData,
 } from '../../src/shared/configUtils.js';
+
 import { ConfigVars } from '../../src/configMeta.js';
 
 describe('configUtils', () => {
   const $$ = new TestContext();
+  const fakeIdentityToken = 'PFT1vw8v65aXd2b9HFvZ3Zu4OcKZwjI60bq7BEjj5k4=';
+  const username = 'SalesforceDeveloper';
 
   afterEach(() => {
     $$.restore();
   });
 
-  it('getOrCreateIdentityToken resolves if token is found', async () => {
-    const fakeIdentityToken = 'fake identity token';
-    $$.SANDBOX.stub(ConfigUtils, 'getIdentityToken').resolves(fakeIdentityToken);
+  it('getOrCreateIdentityToken resolves if identity data is found', async () => {
+    const identityData = new LocalWebServerIdentityData(fakeIdentityToken);
+    identityData.usernameToServerEntityIdMap[username] = 'entityId';
+    const stubConnection = $$.SANDBOX.createStubInstance(Connection);
+    $$.SANDBOX.stub(ConfigUtils, 'getIdentityData').resolves(identityData);
+    $$.SANDBOX.stub(Connection, 'create').resolves(Connection.prototype);
+
+    const resolved = await ConfigUtils.getOrCreateIdentityToken(username, stubConnection);
 
-    const resolved = await ConfigUtils.getOrCreateIdentityToken();
     expect(resolved).to.equal(fakeIdentityToken);
   });
 
-  it('getOrCreateIdentityToken resolves and writeIdentityToken is called when there is no token', async () => {
-    const fakeIdentityToken = 'fake identity token';
-    $$.SANDBOX.stub(ConfigUtils, 'getIdentityToken').resolves(undefined);
+  it('getOrCreateIdentityToken resolves and writeIdentityData is called when there is no identity data', async () => {
+    const stubConnection = $$.SANDBOX.createStubInstance(Connection);
+    $$.SANDBOX.stub(ConfigUtils, 'getIdentityData').resolves(undefined);
     $$.SANDBOX.stub(CryptoUtils, 'generateIdentityToken').resolves(fakeIdentityToken);
-    const writeIdentityTokenStub = $$.SANDBOX.stub(ConfigUtils, 'writeIdentityToken').resolves();
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    $$.SANDBOX.stub(ConfigUtils as any, 'saveIdentityTokenToServer')
```

Comment:
Had to cast `ConfigUtils` as `any` because `saveIdentityTokenToServer` is a private static method. Without the cast sinon has difficulty stubbing a private method.

Reply:
Instead of stubbing a private function, you could stub `sobject` and `insert` on `Connection.prototype`